docs/contribute/html-gui.md

```diff
@@ -0,0 +1,126 @@
+In order to build and test Saros it was necessary to download an OS specific SWT version (this was handeled by the build tool). It was easy to
+release the solution for eclipse, because eclipse already provides a swt version, because the whole IDE is written in swt.
+
+### IntelliJ integration
+The IntelliJ version of the integration required to load a swt version manually which was loaded during runtime of Saros with a self-written
+SWT loader. This is not a solution that is safe to be released. The whole IntelliJ specific code was located in the package `saros.intellij.ui.swt_browser`.
+
+### Issues with the integration
+Even if the SWT library was available and loaded, the browser integration was brittle.
+This was rooted in the way the SWT browser was implemented. Instead of releasing an browser with swt, the OS default browser is used.
+This integration of the default browser was brittle on all platforms, but the most unstable was Linux. The newer versions of swt (e.g. swt released with eclipse 4.8) are more stable, but the `BrowserFunction` (allows to call java from javascript) was still not stable in all versions (unavailable in 4.8-4.12 due to [#538335](https://bugs.eclipse.org/bugs/show_bug.cgi?id=538335)).
+
+## JavaFX
+The second approach was to embed a view of the GUI toolkit JavaFX scene into the view of the Saros plugin. JavaFX provides a WebView with integrated WebKit version.
+Furthermore, JavaFX provides a bridge to swt as well as to swing. Therefore it is possible to embed a JavaFX component into an eclipse plugin (that uses swt) as well
+as into an IntelliJ plugin (that uses swing). 
+
+### Eclipse integration
+We integrated JavaFX in eclipse with the help of the efxclipse tooling and integrated the feature `org.eclipse.fx.target.rcp.feature`.
+This feature provides the osgi bundle `org.eclipse.fx.ui.workbench3`. This bundle provides the class `org.eclipse.fx.ui.workbench3.FXPartView` that allows to write an e3 (old eclipse ui model) view with JavaFX components.
+This class simply wraps the `javafx.embed.swt.FXCanvas`. In order to load the JavaFX library during runtime the osgi bundle `org.eclipse.fx.osgi` is also provided. This bundle is a fragment of the bundle `org.eclipse.osgi` and adds a hook that enhances the classpath of eclipse in order to load the additional javafx/swt jar that allows to integrate javafx components into swt. If this bundle is missing, the class `javafx.embed.swt.FXCanvas` is missing.
+
+### IntelliJ integration
+The integration into IntelliJ was straightforward: create and use a `javafx.embed.swing.JFXPanel`. The following [post](https://stackoverflow.com/a/35611230/6948317) gives an implementation example.
+
+### Eclipse tests
+The first tests on Windows and Linux were successful (tested with Eclipse Neon, Oxygen, 2019-06 and JDK8-9). The only issue
+was that it was necessary with JDK8 to force eclipse to use GTK 3, because the JavaFX library (shipped with JDK8) was linked to GTK 2.
```

Comment:
*forced not to use* or *forced to use GTK 2*. Forcing GTK3 would cause the crash.

Reply:
:see_no_evil: True.

Reply:
Done